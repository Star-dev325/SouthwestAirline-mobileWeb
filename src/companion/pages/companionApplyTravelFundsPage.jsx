// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import withHideGlobalHeader from 'src/shared/enhancers/withHideGlobalHeader';
import ApplyFundsFormWithLedger from 'src/travelFunds/components/applyFundsFormWithLedger';
import * as TravelFundsActions from 'src/travelFunds/actions/travelFundsActions';
import * as ApplyTravelFundsActions from 'src/shared/actions/applyTravelFundsActions';
import * as DialogActions from 'src/shared/actions/dialogActions';
import * as AnalyticsActions from 'src/shared/analytics/actions/analyticsActions';
import {
  transformToRTFCalculateRequest,
  transformToVoucherCalculateRequest,
  transformToCardCalculateRequest,
  transformToRemoveFundsRequest,
  transformToRefreshFundsRequest
} from 'src/travelFunds/transformers/travelFundsTransformer';
import {
  getPriceTotalWithEBForCompanion,
  getBalanceRemainingWithEBForCompanion
} from 'src/shared/selectors/earlyBirdSelector';
import { getCompanionPassengerInfos } from 'src/companion/selectors/companionPurchaseSummaryPageSelectors';
import TravelFundsConstants from 'src/travelFunds/constants/travelFundsConstants';
import { TRAVEL_FUNDS } from 'src/companion/constants/companionConstants';
import i18n from '@swa-ui/locale';

import type { Push, PassengerInfos, TotalsType } from 'src/shared/flow-typed/shared.types';
import type {
  LookUpRTFFundType,
  LookUpVoucherFundType,
  LookUpCardFundType,
  TravelFundsOptionsType,
  ApplyTravelFundsPageResponse
} from 'src/travelFunds/flow-typed/travelFunds.types';
import type {
  CalcFundsRequestType,
  RemoveFundRequestType,
  RefreshFundsRequestType
} from 'src/airBooking/flow-typed/calcFunds.types';
import type { FormData } from 'src/shared/form/flow-typed/form.types';

const { APPLY_FUNDS_CANCEL_DIALOG, FUND_TYPES_FORMATTED } = TravelFundsConstants;

type Props = {
  push: Push,
  goBack: () => void,
  balanceRemaining?: { totals: TotalsType },
  priceTotal: { totals: ?TotalsType },
  applyTravelFundsPageResponse: ?ApplyTravelFundsPageResponse,
  removeFundFn: (RemoveFundRequestType, ?string, ?boolean) => void,
  calculateFundsFn: (CalcFundsRequestType, ?string, ?boolean) => void,
  updateSelectedApplyTabFn: (TravelFundsOptionsType) => void,
  refreshFundsFn: (RefreshFundsRequestType, ?string, ?boolean) => void,
  resetCalculateFlowDataFn: () => void,
  hideDialogFn: () => Promise<*>,
  showDialogFn: (*) => Promise<*>,
  fundsAppliedToken?: string,
  passengerInfos: PassengerInfos,
  itineraryPricingToken: string,
  isLoggedIn: boolean,
  currentlySelectedTab: TravelFundsOptionsType,
  saveLastSearchedFundFn: (TravelFundsOptionsType, FormData) => void,
  clearAllApplyFormsFn: () => void
};

const { TOKEN_EXPIRED_COMPANION_URL } = TRAVEL_FUNDS;

export class CompanionApplyTravelFundsPage extends Component<Props> {
  componentDidMount() {
    const { refreshFundsFn, passengerInfos, fundsAppliedToken, itineraryPricingToken, isLoggedIn } = this.props;

    fundsAppliedToken &&
      refreshFundsFn(
        transformToRefreshFundsRequest(passengerInfos, fundsAppliedToken, itineraryPricingToken),
        TOKEN_EXPIRED_COMPANION_URL,
        isLoggedIn
      );
  }

  _onSelectionChange = (selection: TravelFundsOptionsType) => {
    const { updateSelectedApplyTabFn } = this.props;

    updateSelectedApplyTabFn(selection);
  };

  _onSubmitRTFCalculate = (formData: LookUpRTFFundType) => {
    const {
      calculateFundsFn,
      passengerInfos,
      fundsAppliedToken,
      itineraryPricingToken,
      isLoggedIn,
      saveLastSearchedFundFn
    } = this.props;

    saveLastSearchedFundFn(FUND_TYPES_FORMATTED[0], formData);
    calculateFundsFn(
      transformToRTFCalculateRequest(formData, passengerInfos, fundsAppliedToken, itineraryPricingToken),
      TOKEN_EXPIRED_COMPANION_URL,
      isLoggedIn
    );
  };

  _onSubmitVoucherCalculate = (formData: LookUpVoucherFundType) => {
    const {
      calculateFundsFn,
      passengerInfos,
      fundsAppliedToken,
      itineraryPricingToken,
      isLoggedIn,
      saveLastSearchedFundFn
    } = this.props;

    saveLastSearchedFundFn(FUND_TYPES_FORMATTED[1], formData);
    calculateFundsFn(
      transformToVoucherCalculateRequest(formData, passengerInfos, fundsAppliedToken, itineraryPricingToken),
      TOKEN_EXPIRED_COMPANION_URL,
      isLoggedIn
    );
  };

  _onSubmitCardCalculate = (formData: LookUpCardFundType) => {
    const {
      calculateFundsFn,
      passengerInfos,
      fundsAppliedToken,
      itineraryPricingToken,
      isLoggedIn,
      saveLastSearchedFundFn
    } = this.props;

    saveLastSearchedFundFn(FUND_TYPES_FORMATTED[2], formData);
    calculateFundsFn(
      transformToCardCalculateRequest(formData, passengerInfos, fundsAppliedToken, itineraryPricingToken),
      TOKEN_EXPIRED_COMPANION_URL,
      isLoggedIn
    );
  };

  _removeTravelFund = (fundId: string) => {
    const {
      removeFundFn,
      passengerInfos,
      fundsAppliedToken,
      itineraryPricingToken,
      resetCalculateFlowDataFn,
      applyTravelFundsPageResponse,
      isLoggedIn
    } = this.props;
    const travelFunds = _.get(applyTravelFundsPageResponse, 'travelFunds', []);

    travelFunds.length > 1 &&
      removeFundFn(
        transformToRemoveFundsRequest(fundId, passengerInfos, fundsAppliedToken, itineraryPricingToken),
        TOKEN_EXPIRED_COMPANION_URL,
        isLoggedIn
      );
    travelFunds.length === 1 && resetCalculateFlowDataFn();
  };

  _onClickCancelButton = () => {
    const {
      applyTravelFundsPageResponse,
      hideDialogFn,
      showDialogFn,
      goBack,
      clearAllApplyFormsFn,
      resetCalculateFlowDataFn
    } = this.props;
    const travelFunds = _.get(applyTravelFundsPageResponse, 'travelFunds', null);

    travelFunds &&
      showDialogFn({
        name: APPLY_FUNDS_CANCEL_DIALOG.name,
        title: APPLY_FUNDS_CANCEL_DIALOG.title,
        message: APPLY_FUNDS_CANCEL_DIALOG.message,
        buttons: [
          {
            label: i18n('SHARED__BUTTON_TEXT__CANCEL'),
            onClick: hideDialogFn
          },
          {
            label: i18n('SHARED__BUTTON_TEXT__OK'),
            onClick: () => {
              hideDialogFn().then(() => {
                goBack();
                clearAllApplyFormsFn();
                resetCalculateFlowDataFn();
              });
            }
          }
        ]
      });

    if (!travelFunds) {
      clearAllApplyFormsFn();
      goBack();
    }
  };

  _returnToPurchasePage = () => {
    const { goBack, clearAllApplyFormsFn } = this.props;

    clearAllApplyFormsFn();
    goBack();
  };

  render() {
    const { currentlySelectedTab, applyTravelFundsPageResponse, priceTotal, balanceRemaining } = this.props;
    const travelFundsBalanceRemaining = _.get(balanceRemaining, 'totals.moneyTotal', null);

    return (
      <ApplyFundsFormWithLedger
        currentlySelectedTab={currentlySelectedTab}
        travelFundsBalanceRemaining={travelFundsBalanceRemaining}
        applyTravelFundsPageResponse={applyTravelFundsPageResponse}
        priceTotal={priceTotal}
        applyFundsToPurchaseFn={this._returnToPurchasePage}
        onSubmitRTFCalculateFn={this._onSubmitRTFCalculate}
        onSubmitVoucherCalculateFn={this._onSubmitVoucherCalculate}
        onClickCancelButtonFn={this._onClickCancelButton}
        onSelectionChangeFn={this._onSelectionChange}
        onSubmitCardCalculateFn={this._onSubmitCardCalculate}
        removeTravelFundFn={this._removeTravelFund}
      />
    );
  }
}

const mapStateToProps = (state) => ({
  passengerInfos: getCompanionPassengerInfos(state),
  applyTravelFundsPageResponse: _.get(state, 'app.companion.applyTravelFundsPage.response'),
  itineraryPricingToken: _.get(
    state,
    'app.companion.flightPricingPage._links.calculateFunds.body.itineraryPricingToken'
  ),
  currentlySelectedTab: _.get(state, 'app.companion.applyTravelFundsPage.currentlySelectedTab') || 'travel-funds',
  fundsAppliedToken: _.get(state, 'app.companion.applyTravelFundsPage.response.fundsAppliedToken'),
  isLoggedIn: _.get(state, 'app.account.isLoggedIn'),
  priceTotal: getPriceTotalWithEBForCompanion(state),
  balanceRemaining: getBalanceRemainingWithEBForCompanion(state)
});

const mapDispatchToProps = {
  calculateFundsFn: ApplyTravelFundsActions.calculateFunds,
  removeFundFn: ApplyTravelFundsActions.removeFund,
  refreshFundsFn: ApplyTravelFundsActions.refreshFunds,
  resetCalculateFlowDataFn: ApplyTravelFundsActions.resetCalculateFlowData,
  updateSelectedApplyTabFn: TravelFundsActions.updateSelectedApplyTab,
  clearAllApplyFormsFn: TravelFundsActions.clearAllApplyForms,
  showDialogFn: DialogActions.showDialog,
  hideDialogFn: DialogActions.hideDialog,
  saveLastSearchedFundFn: AnalyticsActions.saveLastSearchedFund
};

const enhancers = _.flowRight(
  withConnectedReactRouter,
  withHideGlobalHeader,
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(CompanionApplyTravelFundsPage);
