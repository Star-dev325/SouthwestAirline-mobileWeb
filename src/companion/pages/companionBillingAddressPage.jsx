// @flow
import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

import * as CompanionActions from 'src/companion/actions/companionActions';
import BillingAddressForm from 'src/shared/components/billingAddressForm';
import withHideGlobalHeader from 'src/shared/enhancers/withHideGlobalHeader';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import { transformContactInfoToBillingAddressFormData } from 'src/shared/transformers/billingAddressTransformer';
import { COMPANION_BILLING_ADDRESS_FORM } from 'src/shared/constants/formIds';

import type { Push, AccountContactInfoType, BillingAddressFormType } from 'src/shared/flow-typed/shared.types';
import type { FormData } from 'src/shared/form/flow-typed/form.types';

type Props = {
  goBack: () => void,
  push: Push,
  travelFundsAddress?: BillingAddressFormType,
  contactInfo?: AccountContactInfoType,
  saveTravelFundsBillingAddressFn: (BillingAddressFormType) => void
};

export class CompanionBillingAddressPage extends React.Component<Props> {
  _buildInitialFormData = () => {
    const { travelFundsAddress, contactInfo } = this.props;

    if (_.isEmpty(travelFundsAddress) && contactInfo) {
      return transformContactInfoToBillingAddressFormData(contactInfo);
    } else if (travelFundsAddress) {
      return travelFundsAddress;
    }
  };

  _onSubmit = (billingAddressFormData: FormData) => {
    const { saveTravelFundsBillingAddressFn, goBack } = this.props;

    saveTravelFundsBillingAddressFn(billingAddressFormData);
    goBack();
  };

  render() {
    const initialFormData = this._buildInitialFormData();

    return (
      <div>
        <BillingAddressForm
          initialFormData={initialFormData}
          formId={COMPANION_BILLING_ADDRESS_FORM}
          onSubmit={this._onSubmit}
          goBack={this.props.goBack}
        />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  travelFundsAddress: _.get(state, 'app.companion.travelFundsAddress'),
  contactInfo: _.get(state, 'app.account.accountInfo.contactInfo')
});

const mapDispatchToProps = {
  saveTravelFundsBillingAddressFn: CompanionActions.saveTravelFundsBillingAddress
};

const enhancers = _.flowRight(
  withHideGlobalHeader,
  withConnectedReactRouter,
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(CompanionBillingAddressPage);
