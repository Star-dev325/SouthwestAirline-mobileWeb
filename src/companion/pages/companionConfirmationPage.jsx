// @flow
import _ from 'lodash';
import { connect } from 'react-redux';
import { getCompanionConfirmationPage } from 'src/companion/selectors/companionConfirmationPageSelector';
import * as CarBookingActions from 'src/carBooking/actions/carBookingActions';
import CarCrossSellBanner from 'src/airBooking/components/carCrossSellBanner';
import ConfirmationTripHeader from 'src/shared/components/confirmationTripHeader';
import Container from 'src/shared/components/container';
import FundResultsList from 'src/shared/components/fundResultsList';
import i18n from '@swa-ui/locale';
import MessageWithInstructions from 'src/shared/components/messageWithInstructions';
import NavGroupItemLinks from 'src/shared/components/navGroupItemLinks';
import PriceTotal from 'src/shared/components/priceTotal';
import React from 'react';
import ReservationFlightSummary from 'src/shared/components/reservationFlightSummary';
import SubHeader from 'src/shared/components/subHeader';
import type { BillingInformation, ConfirmationDates, ConfirmationPassengers, FlightPricingBound, Push, TotalsType } from 'src/shared/flow-typed/shared.types';
import type { GetCarBookingLinkQueryType } from 'src/carBooking/flow-typed/carBooking.types';
import type { RetrievedFundType } from 'src/travelFunds/flow-typed/travelFunds.types';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';

type Props = {
  _links: { carBooking?: Link },
  billingInfo: BillingInformation,
  bounds: Array<FlightPricingBound>,
  dates: ConfirmationDates,
  destinationDescription: string,
  fundsApplied?: Array<RetrievedFundType>,
  headerMessage?: { body: string, header: string, key: string },
  prepareCarCrossSellFromQueryAndTransitionToCarBookingFn: (carBookingLinkQuery: GetCarBookingLinkQueryType) => void,
  pnrs: Array<ConfirmationPassengers>,
  push: Push,
  totals: TotalsType
};

export class CompanionConfirmationPage extends React.Component<Props> {
  _onCarCrossSellClick = () => {
    const { prepareCarCrossSellFromQueryAndTransitionToCarBookingFn, _links } = this.props;

    prepareCarCrossSellFromQueryAndTransitionToCarBookingFn(_.get(_links, 'carBooking.query'));
  };

  render() {
    const { dates, destinationDescription, pnrs, bounds, totals, billingInfo, fundsApplied, headerMessage } =
      this.props;

    return (
      <div className="companion-confirmation">
        <SubHeader title="Confirmation" />

        <MessageWithInstructions
          className="purchase-confirmation--trip-booked"
          title={headerMessage ? headerMessage.header : i18n('COMPANION_MESSAGE_WITH_INSTRUCTIONS_TITLE')}
          subInstruction={
            headerMessage
              ? headerMessage.body
              : _.isEmpty(bounds)
                ? i18n('SHARED__TRIP_BOOKED__EMAIL_CONFIRMATION_SUB_DETAILS')
                : i18n('COMPANION_MESSAGE_WITH_INSTRUCTIONS_SUB_INSTRUCTION')
          }
        />
        <ConfirmationTripHeader
          dates={dates}
          destinationDescription={destinationDescription}
          pnrs={pnrs}
          bounds={bounds}
        />
        <ReservationFlightSummary bounds={bounds} />
        <PriceTotal totals={totals} />
        <FundResultsList
          listTitle={i18n('SHARED__TRIP_BOOKED__AMOUNT_APPLIED')}
          billingInfo={billingInfo}
          retrievedFunds={fundsApplied}
        />
        <Container noBottomPadding>
          <CarCrossSellBanner onClick={this._onCarCrossSellClick} />
          <NavGroupItemLinks />
        </Container>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  ...getCompanionConfirmationPage(state)
});

const mapDispatchToProps = {
  saveSelectedRecentSearchRequestFn: CarBookingActions.saveSelectedRecentSearchRequest,
  prepareCarCrossSellFromQueryAndTransitionToCarBookingFn:
    CarBookingActions.prepareCarCrossSellFromQueryAndTransitionToCarBooking
};

const enhancers = _.flowRight(withConnectedReactRouter, connect(mapStateToProps, mapDispatchToProps));

export default enhancers(CompanionConfirmationPage);
