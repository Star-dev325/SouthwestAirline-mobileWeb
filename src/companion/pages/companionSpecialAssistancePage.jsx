// @flow
import React from 'react';
import _ from 'lodash';
import { connect } from 'react-redux';

import * as FormDataActions from 'src/shared/actions/formDataActions';
import * as CompanionActions from 'src/companion/actions/companionActions';
import withHideGlobalHeader from 'src/shared/enhancers/withHideGlobalHeader';
import withConnectedReactRouter from 'src/shared/enhancers/withConnectedReactRouter';
import SpecialAssistancePage from 'src/shared/pages/specialAssistancePage';
import { COMPANION_SPECIAL_ASSISTANCE_FORM } from 'src/shared/constants/formIds';
import { DEFAULT_FIELD_VALUES } from 'src/shared/constants/specialAssistanceConstants';

import type { SpecialAssistanceType } from 'src/shared/flow-typed/shared.types';

type Props = {
  goBack: () => void,
  updateFormDataValueFn: (string, *) => {},
  clearFormDataByIdFn: (string) => {},
  updateCompanionWithSpecialAssistanceFn: (*) => void,
  specialAssistanceFormData?: *,
  savedFormData?: SpecialAssistanceType
};

export class CompanionSpecialAssistancePage extends React.Component<Props> {
  _onFormSubmit = () => {
    const { specialAssistanceFormData, updateCompanionWithSpecialAssistanceFn, clearFormDataByIdFn } = this.props;

    updateCompanionWithSpecialAssistanceFn(specialAssistanceFormData);
    clearFormDataByIdFn(COMPANION_SPECIAL_ASSISTANCE_FORM);
  };

  render() {
    const { updateFormDataValueFn, savedFormData, goBack } = this.props;
    const initialFormData = _.merge({}, DEFAULT_FIELD_VALUES, savedFormData);

    return (
      <SpecialAssistancePage
        onSubmit={this._onFormSubmit}
        formId={COMPANION_SPECIAL_ASSISTANCE_FORM}
        goBack={goBack}
        initialFormData={initialFormData}
        updateFormDataValueFn={updateFormDataValueFn}
      />
    );
  }
}

const mapStateToProps = (state) => ({
  specialAssistanceFormData: _.get(state, `app.formData.${COMPANION_SPECIAL_ASSISTANCE_FORM}.data`),
  savedFormData: _.get(state, `app.companion.specialAssistance`)
});

const mapDispatchToProps = {
  updateCompanionWithSpecialAssistanceFn: CompanionActions.updateCompanionWithSpecialAssistance,
  updateFormDataValueFn: FormDataActions.updateFormDataValue,
  clearFormDataByIdFn: FormDataActions.clearFormDataById
};

const enhancers = _.flowRight(
  withConnectedReactRouter,
  withHideGlobalHeader,
  connect(mapStateToProps, mapDispatchToProps)
);

export default enhancers(CompanionSpecialAssistancePage);
