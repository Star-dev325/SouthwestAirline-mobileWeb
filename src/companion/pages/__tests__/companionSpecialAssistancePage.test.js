import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { CompanionSpecialAssistancePage } from 'src/companion/pages/companionSpecialAssistancePage';
import { COMPANION_SPECIAL_ASSISTANCE_FORM } from 'src/shared/constants/formIds';

describe('CompanionSpecialAssistancePage', () => {
  let clearFormDataByIdFnStub;
  let goBackStub;
  let updateCompanionWithSpecialAssistanceFnStub;

  beforeEach(() => {
    goBackStub = jest.fn();
    updateCompanionWithSpecialAssistanceFnStub = jest.fn();
    clearFormDataByIdFnStub = jest.fn();
  });
  describe('should render', () => {
    it('SpecialAssistancePage', () => {
      const { container } = createComponent();

      expect(container).toMatchSnapshot();
    });

    it('SpecialAssistanceForm', () => {
      const { container } = createComponent();

      expect(container.querySelector('.special-assistance-form')).not.toBeNull();
    });
  });

  describe('on submit', () => {
    it('should save companion passenger info and back to purchase summary page when submit', () => {
      const { container } = createComponent({ specialAssistanceFormData: { BLIND: true } });

      fireEvent.submit(container.querySelector('form'));

      expect(updateCompanionWithSpecialAssistanceFnStub).toHaveBeenCalledWith({ BLIND: true });
      expect(clearFormDataByIdFnStub).toHaveBeenCalledWith(COMPANION_SPECIAL_ASSISTANCE_FORM);
    });
  });

  const createComponent = (props = {}) => {
    const noop = () => {};
    const defaultProps = {
      goBack: goBackStub,
      updateCompanionWithSpecialAssistanceFn: updateCompanionWithSpecialAssistanceFnStub,
      updateFormDataValueFn: noop,
      clearFormDataByIdFn: clearFormDataByIdFnStub,
      specialAssistanceFormData: {},
      savedFormData: {}
    };

    const store = configureMockStore()({
      app: {
        errorHeader: {
          hasError: false,
          errorMessage: null
        }
      },
      router: {
        location: {
          search: 'search'
        }
      }
    });

    return render(
      <Provider store={store}>
        <CompanionSpecialAssistancePage {...defaultProps} {...props} />
      </Provider>
    );
  };
});
