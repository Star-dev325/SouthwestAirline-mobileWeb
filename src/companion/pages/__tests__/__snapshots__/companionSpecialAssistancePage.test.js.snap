// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanionSpecialAssistancePage should render SpecialAssistancePage 1`] = `
<div>
  <form
    class="special-assistance-form form"
    name="COMPANION_SPECIAL_ASSISTANCE_FORM"
    novalidate=""
  >
    <fieldset>
      <div>
        <div
          class="page-header action-bar"
          id=""
        >
          <ul
            class="action-bar-buttons action-bar--left-buttons"
          />
          <div
            class="action-bar--container"
          >
            <div
              class="action-bar--title"
            >
              <span
                class="action-bar--main-title"
              >
                SHARED__SPECIAL_ASSISTANCE__SA_PAGE_TITLE
              </span>
              <span
                class="action-bar--sub-title"
              />
            </div>
          </div>
          <ul
            class="action-bar-buttons action-bar--right-buttons"
          >
            <li
              class="action-bar-buttons--item"
            >
              <button
                class="button"
                type="button"
              >
                SHARED__BUTTON_TEXT__CANCEL
              </button>
            </li>
          </ul>
        </div>
        <div />
      </div>
      <div
        class="segments"
      >
        <div
          class="secondary fill segment"
        >
          <p
            class="helper-text"
          >
            SHARED__SPECIAL_ASSISTANCE__SA_HELP_TEXT
          </p>
          <p
            class="helper-text"
          >
            <a
              href="/special-assistance-info"
              target="_blank"
            >
              Learn more
            </a>
          </p>
          <div
            class="fields sa-checkbox-fields divided grouped"
          >
            <label
              class="fields--label"
            >
              SHARED__SPECIAL_ASSISTANCE__AIRPORT_ASSISTANCE_SEGMENT_LABEL
            </label>
            <div
              class="field"
            >
              <div
                class="checkbox-button relative flex flex-cross-center"
                name="BLIND"
              >
                <div
                  class="flex1 flex-cross-center"
                >
                  <span
                    class="checkbox-button--mark"
                  >
                    <i
                      class="absolute xlarge bgtransp icon icon_check"
                    />
                  </span>
                </div>
                <div
                  class="flex10 ml0"
                >
                  <div
                    class="checkbox-button--children flex-equal-width ml2"
                  >
                    SHARED__SPECIAL_ASSISTANCE__AIRPORT_ASSISTANCE_BLIND
                  </div>
                </div>
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="checkbox-button relative flex flex-cross-center"
                name="DEAF"
              >
                <div
                  class="flex1 flex-cross-center"
                >
                  <span
                    class="checkbox-button--mark"
                  >
                    <i
                      class="absolute xlarge bgtransp icon icon_check"
                    />
                  </span>
                </div>
                <div
                  class="flex10 ml0"
                >
                  <div
                    class="checkbox-button--children flex-equal-width ml2"
                  >
                    SHARED__SPECIAL_ASSISTANCE__AIRPORT_ASSISTANCE_DEAF
                  </div>
                </div>
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="checkbox-button relative flex flex-cross-center"
                name="COGNITIVE_AND_DEVELOPMENTAL_SSR"
              >
                <div
                  class="flex1 flex-cross-center"
                >
                  <span
                    class="checkbox-button--mark"
                  >
                    <i
                      class="absolute xlarge bgtransp icon icon_check"
                    />
                  </span>
                </div>
                <div
                  class="flex10 ml0"
                >
                  <div
                    class="checkbox-button--children flex-equal-width ml2"
                  >
                    SHARED__SPECIAL_ASSISTANCE__AIRPORT_ASSISTANCE_COGNITIVE_AND_DEVELOPMENTAL_SSR
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="secondary fill segment"
        >
          <div
            class="fields sa-checkbox-fields divided grouped"
          >
            <label
              class="fields--label"
            >
              SHARED__SPECIAL_ASSISTANCE__ANIMALS__SEGMENT_LABEL
            </label>
            <div
              class="field"
            >
              <div
                class="checkbox-button relative flex flex-cross-center"
                name="ASSISTANCE_ANIMAL"
              >
                <div
                  class="flex1 flex-cross-center"
                >
                  <span
                    class="checkbox-button--mark"
                  >
                    <i
                      class="absolute xlarge bgtransp icon icon_check"
                    />
                  </span>
                </div>
                <div
                  class="flex10 ml0"
                >
                  <div
                    class="checkbox-button--children flex-equal-width ml2"
                  >
                    SHARED__SPECIAL_ASSISTANCE__ASSISTANCE_ANIMAL
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="secondary fill segment"
        >
          <div
            class="fields sa-radio-fields divided grouped"
          >
            <label
              class="fields--label"
            >
              SHARED__SPECIAL_ASSISTANCE__WHEELCHAIR_ASSISTANCE_SEGMENT_LABEL
            </label>
            <div
              class="field"
            >
              <div
                class="sa-radio-options"
              >
                <div
                  class="sa-radio-item sa-radio-item_checked"
                  name="NONE"
                >
                  <div
                    class="sa-radio-item--radio"
                  >
                    <span
                      class="radio-button--mark"
                    >
                      <span
                        class="radio-input--mark active"
                      />
                    </span>
                  </div>
                  <div
                    class="sa-radio-item--text"
                  >
                    No wheelchair assistance needed
                  </div>
                </div>
                <div
                  class="sa-radio-item"
                  name="AIRPORT_WHEELCHAIR"
                >
                  <div
                    class="sa-radio-item--radio"
                  >
                    <span
                      class="radio-button--mark"
                    >
                      <span
                        class="radio-input--mark"
                      />
                    </span>
                  </div>
                  <div
                    class="sa-radio-item--text"
                  >
                    Can walk but need assistance to and from gate
                  </div>
                </div>
                <div
                  class="sa-radio-item"
                  name="AISLE_CHAIR"
                >
                  <div
                    class="sa-radio-item--radio"
                  >
                    <span
                      class="radio-button--mark"
                    >
                      <span
                        class="radio-input--mark"
                      />
                    </span>
                  </div>
                  <div
                    class="sa-radio-item--text"
                  >
                    Need lift/transfer assistance to and from aircraft seat
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="secondary fill segment"
        >
          <div
            class="fields sa-radio-fields divided grouped"
          >
            <label
              class="fields--label"
            >
              PERSONAL WHEELCHAIR STOWAGE
            </label>
            <div
              class="field"
            >
              <div
                class="sa-radio-options"
              >
                <div
                  class="sa-radio-item sa-radio-item_checked"
                  name="NONE"
                >
                  <div
                    class="sa-radio-item--radio"
                  >
                    <span
                      class="radio-button--mark"
                    >
                      <span
                        class="radio-input--mark active"
                      />
                    </span>
                  </div>
                  <div
                    class="sa-radio-item--text"
                  >
                    No wheelchair stowage needed
                  </div>
                </div>
                <div
                  class="sa-radio-item"
                  name="MANUAL_WHEELCHAIR"
                >
                  <div
                    class="sa-radio-item--radio"
                  >
                    <span
                      class="radio-button--mark"
                    >
                      <span
                        class="radio-input--mark"
                      />
                    </span>
                  </div>
                  <div
                    class="sa-radio-item--text"
                  >
                    Manual wheelchair
                  </div>
                </div>
                <div
                  class="sa-radio-item sa-radio-item_select"
                  name="WET_CELL_BATTERY_WHEELCHAIR"
                >
                  <div
                    class="sa-radio-item--radio"
                  >
                    <span
                      class="radio-button--mark"
                    >
                      <span
                        class="radio-input--mark"
                      />
                    </span>
                  </div>
                  <div
                    class="sa-radio-item--text"
                  >
                    Powered wheelchair (how many spillable batteries?)
                  </div>
                </div>
                <div
                  class="sa-radio-item sa-radio-item_select"
                  name="DRY_CELL_BATTERY_WHEELCHAIR"
                >
                  <div
                    class="sa-radio-item--radio"
                  >
                    <span
                      class="radio-button--mark"
                    >
                      <span
                        class="radio-input--mark"
                      />
                    </span>
                  </div>
                  <div
                    class="sa-radio-item--text"
                  >
                    Powered wheelchair (how many non-spillable batteries?)
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="secondary fill segment"
        >
          <div
            class="fields sa-checkbox-fields divided grouped"
          >
            <label
              class="fields--label"
            >
              SHARED__SPECIAL_ASSISTANCE__OTHER_ASSISTANCE_SEGMENT_LABEL
            </label>
            <div
              class="field"
            >
              <div
                class="checkbox-button relative flex flex-cross-center"
                name="PEANUT_DUST_ALLERGY"
              >
                <div
                  class="flex1 flex-cross-center"
                >
                  <span
                    class="checkbox-button--mark"
                  >
                    <i
                      class="absolute xlarge bgtransp icon icon_check"
                    />
                  </span>
                </div>
                <div
                  class="flex10 ml0"
                >
                  <div
                    class="checkbox-button--children flex-equal-width ml2"
                  >
                    SHARED__SPECIAL_ASSISTANCE__OTHER_ASSISTANCE_PEANUT_DUST_ALLERGY
                  </div>
                </div>
              </div>
            </div>
            <div
              class="field"
            >
              <div
                class="checkbox-button relative flex flex-cross-center"
                name="PORTABLE_OXYGEN_CONCENTRATOR"
              >
                <div
                  class="flex1 flex-cross-center"
                >
                  <span
                    class="checkbox-button--mark"
                  >
                    <i
                      class="absolute xlarge bgtransp icon icon_check"
                    />
                  </span>
                </div>
                <div
                  class="flex10 ml0"
                >
                  <div
                    class="checkbox-button--children flex-equal-width ml2"
                  >
                    SHARED__SPECIAL_ASSISTANCE__OTHER_ASSISTANCE_PORTABLE_OXYGEN_CONCENTRATOR
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="blue inverted segment"
          color="blue"
        >
          <button
            class="button button--fluid huge button--yellow"
            type="submit"
          >
            SHARED__BUTTON_TEXT__SAVE
          </button>
        </div>
      </div>
    </fieldset>
  </form>
</div>
`;
